thor/orbits/attribution.py

```diff
@@ -40,30 +46,34 @@ def attribution_worker(
     observations,
     eps=1 / 3600,
     include_probabilistic=True,
-    backend="PYOORB",
-    backend_kwargs={},
+    propagator: Literal["PYOORB"] = "PYOORB",
+    propagator_kwargs: dict = {},
 ):
+    if propagator == "PYOORB":
+        prop = PYOORB(**propagator_kwargs)
+    else:
+        raise ValueError(f"Invalid propagator '{propagator}'.")
 
-    # Create observer's dictionary from observations
-    observers = {}
+    # Create Observers table
+    observers_list = []
     for observatory_code in observations["observatory_code"].unique():
-        observers[observatory_code] = Time(
-            observations[observations["observatory_code"].isin([observatory_code])][
-                "mjd_utc"
-            ].unique(),
-            scale="utc",
-            format="mjd",
+        observers_list.append(
+            Observers.from_code(
+                observatory_code,
+                Timestamp.from_mjd(
+                    observations[observations["observatory_code"].isin([observatory_code])]["mjd_utc"].unique(),
+                    scale="utc",
+                )
+            )
         )
 
+    observers = qv.concatenate(observers_list)
+    observers = observers.sort_by(["coordinates.time.days", "coordinates.time.nanos", "code"])
+
     # Genereate ephemerides for each orbit at the observation times
-    ephemeris = generateEphemeris(
-        orbits,
-        observers,
-        backend=backend,
-        backend_kwargs=backend_kwargs,
-        num_jobs=1,
-        chunk_size=1,
-    )
+    ephemeris = prop._generate_ephemeris(orbits, observers)
```

Comment:
Why do we use the private method here?

Reply:
No good reason, we were calling generate_ephemeris (original function) with num_jobs=1 which would have called the private function in series. I'll go through and replace the private calls throughout IOD, OD, and the arc extension code.